packages/types/src/primitive/StorageKey.ts

```diff
@@ -26,27 +32,50 @@ export interface StorageFunction {
  */
 export default class StorageKey extends Bytes {
   private _meta?: MetaV4;
+  private _method?: string;
   private _outputType?: string;
+  private _section?: string;
 
   constructor (value?: AnyU8a | StorageKey | StorageFunction | [StorageFunction, any]) {
-    super(StorageKey.decodeStorageKey(value));
+    const { key, method, section } = StorageKey.decodeStorageKey(value);
+
+    super(key);
 
     this._meta = StorageKey.getMeta(value as StorageKey);
+    this._method = method;
     this._outputType = StorageKey.getType(value as StorageKey);
+    this._section = section;
   }
 
-  static decodeStorageKey (value?: AnyU8a | StorageKey | StorageFunction | [StorageFunction, any]): Uint8Array | string | undefined {
-    if (!value || isU8a(value) || isString(value)) {
+  static decodeStorageKey (value?: AnyU8a | StorageKey | StorageFunction | [StorageFunction, any]): Decoded {
+    if (!value || isString(value)) {
       // let Bytes handle these inputs
-      return value;
+      return {
+        key: value
+      };
+    } else if (isU8a(value)) {
```

Comment:
Totally equivalent, but I would put `instanceof StorageKey` here, and isU8a in the above case (which must then go below in the if/else block).